psycopg/psycopg/_connection_base.py

```diff
@@ -293,8 +293,16 @@ def fileno(self) -> int:
 
     def cancel(self) -> None:
         """Cancel the current operation on the connection."""
-        if self._should_cancel():
-            self._try_cancel(self.pgconn)
+        if not self._should_cancel():
+            return
+
+        # Don't fail cancelling (which might happen on connection closing) to
+        # avoid clobbering eventual exceptions with ours, which is less important.
+        try:
+            c = self.pgconn.get_cancel()
+            c.cancel()
+        except Exception as ex:
+            logger.warning("couldn't try to cancel query: %s", str(ex))
```

Comment:
```suggestion
            logger.warning("couldn't try to cancel query: %s", ex)
```
seems better to me; by avoiding explicit `str()`, which will be done when the log message gets formatted in any case,  we save it if the log level is higher than `WARNING`. (Sure, that's not a critical part; more a nit :)